l5kit/l5kit/evaluation/metrics.py

```diff
@@ -303,3 +305,131 @@ def final_displacement_error_mean(
     """
 
     return _final_displacement_error(ground_truth, pred, confidences, avails, "mean")
+
+# TODO(perone): these functions were moved from the closed-loop evaluator.
+# The functions _ego_agent_within_range, _get_boundingbox, _get_sides would ideally
+# be moved to an abstraction of the Agent later. Such as in the example below:
+#
+#  ego = Agent(...)
+#  agent = Agent(...)
+#  bbox = ego.get_boundingbox()
+#  within_range = ego.within_range(agent)
+#  sides = ego.get_sides()
+
+
+def _ego_agent_within_range(ego_centroid: np.ndarray, ego_extent: np.ndarray,
+                            agent_centroid: np.ndarray, agent_extent: np.ndarray) -> np.ndarray:
+    """This function will check if the agent is within range of the ego. It accepts
+    as input a vectorized form with shapes N,D or a flat vector as well with shapes just D.
+
+    :param ego_centroid: the ego centroid (shape: N, 2)
+    :param ego_extent: the ego extent (shape: N, 3)
+    :param agent_centroid: the agent centroid (shape: N, 2)
+    :param agent_extent: the agent extent (shape: N, 3)
+    :return: array with True if within range, False otherwise (shape: N)
+    """
+    distance = np.linalg.norm(ego_centroid - agent_centroid, axis=-1)
+    norm_ego = np.linalg.norm(ego_extent[..., :2], axis=-1)
+    norm_agent = np.linalg.norm(agent_extent[..., :2], axis=-1)
+    max_range = 0.5 * (norm_ego + norm_agent)
+    return distance < max_range
+
+
+def _get_bounding_box(centroid: np.ndarray, yaw: np.ndarray,
```

Comment:
`/planning/utils.py` already has these functions as we're using collisions in a planning notebook. Should we remove them and fix the notebook to use these?

I can also do it on top of this PR

Reply:
Wasn't aware they were on l5kit as well, I'll check if they have testing/docs there and move these into the `utils.py` in case they don't. I think it would be nice to have a module separate only for geometrical operations like these, also because they aren't specific for planning only.